tests/base.go

```diff
@@ -73,78 +64,80 @@ func RemoteRun(user string, addr string, cmd string) (string, error) {
 	// Connect
 	client, err := ssh.Dial("tcp", net.JoinHostPort(addr, "22"), config)
 	if err != nil {
-		return "", err
+		return "", errors.Wrapf(err, "error starting ssh connection ")
 	}
 	// Create a session. It is one session per command.
 	session, err := client.NewSession()
 	if err != nil {
-		return "", err
+		return "", errors.Wrapf(err, "error creating new session")
 	}
 	defer session.Close()
 	var b bytes.Buffer  // import "bytes"
 	session.Stdout = &b // get output
 	err = session.Run(cmd)
-	return b.String(), err
+	if err != nil {
+		return "", errors.Wrapf(err, "error excuting command on remote")
+	}
+	return b.String(), nil
 }
 
-func VerifyIPs(wgConfig string, verifyIPs []string) bool {
-	UpWg(wgConfig)
+// TODO: investigate if this is needed
+func VerifyIPs(wgConfig string, verifyIPs []string, sk string) error {
 
 	for i := 0; i < len(verifyIPs); i++ {
-		out, _ := exec.Command("ping", verifyIPs[i], "-c 5", "-i 3", "-w 10").Output()
+		out, err := exec.Command("ping", verifyIPs[i], "-c 5", "-i 3", "-w 10").Output()
 		if strings.Contains(string(out), "Destination Host Unreachable") {
-			return false
+			return errors.Wrapf(err, "error host unreachable")
 		}
 	}
-
 	for i := 0; i < len(verifyIPs); i++ {
-		res, _ := RemoteRun("root", verifyIPs[i], "ifconfig")
+		res, err := RemoteRun("root", verifyIPs[i], "ifconfig", sk)
+		if err != nil {
+			return errors.Wrapf(err, "error connecting to ip")
+		}
 		if !strings.Contains(string(res), verifyIPs[i]) {
-			return false
+			return errors.Wrapf(err, "error verifying  ips")
 		}
 	}
-	return true
+	return nil
 }
 
-func RandomName() string {
-	seed := time.Now().UTC().UnixNano()
-	nameGenerator := namegenerator.NewNameGenerator(seed)
-
-	name := nameGenerator.Generate()
-
-	return name
-}
-
-func Wait(addr string, port string) bool {
+// TODO: investigate if this is needed
+// tries to connect to the provided address and port with time out
+func Wait(addr string, port string) error {
+	var err error
 	for t := time.Now(); time.Since(t) < 3*time.Minute; {
-		_, err := net.DialTimeout("tcp", net.JoinHostPort(addr, "22"), time.Second*12)
+		_, err := net.DialTimeout("tcp", net.JoinHostPort(addr, port), time.Second*12)
 		if err == nil {
-			return true
+			return nil
 		}
 	}
-	return true
+	return errors.Wrapf(err, "couldn't join port")
 }
 
-func SshKeys() {
-	os.Mkdir("/tmp/.ssh", 0755)
-	cmd := exec.Command("ssh-keygen", "-t", "rsa", "-f", "/tmp/.ssh/id_rsa", "-q")
-	stdout, err := cmd.Output()
-
+// creating the public and private key for the machine
+func SshKeys() (string, string, error) {
```

Comment:
This function Generates , so let's Name it `GenerateSSHKeyPair` right? @rawdaGastan please write down how to simplify to a simpler version, I believe we can get away with only the privateKey and call a `.PublicKey` function at anypoint to get the public key

Reply:
'func SshKeys() (string, string, error) {

  rsaKey, err := rsa.GenerateKey(rand.Reader, 1024)
  if err != nil {
    return "", "", errors.Wrapf(err, "error generating rsa key")
  }

  pemKey := &pem.Block{Type: "RSA PRIVATE KEY", Bytes: x509.MarshalPKCS1PrivateKey(rsaKey)}
  privateKey := pem.EncodeToMemory(pemKey)

  pub, err := ssh.NewPublicKey(&rsaKey.PublicKey)
  if err != nil {
    return "", "", errors.Wrapf(err, "error extracting public key")
  }
  authorizedKey := ssh.MarshalAuthorizedKey(pub)
  return string(authorizedKey), string(privateKey), nil
}'